internal/internal_worker.go

```diff
@@ -354,23 +371,83 @@ func (ww *workflowWorker) Stop() {
 	ww.worker.Stop()
 }
 
+func newSessionWorker(service workflowserviceclient.Interface,
+	domain string,
+	params workerExecutionParameters,
+	overrides *workerOverrides,
+	env *hostEnvImpl,
+	maxConCurrentSessionExecutionSize int,
+) Worker {
+	if params.Identity == "" {
+		params.Identity = getWorkerIdentity(params.TaskList)
+	}
+	if params.SessionResourceID == "" {
+		// TODO: Should we generate a uuid for user?
+		params.SessionResourceID = "Resource_" + uuid.New()
```

Comment:
What about using worker identity (with appended uuid if it is not unique across restarts)?

Reply:
Yeah, it's better than hardcoding the "Resource" prefix. Updated.